### PR TITLE
Fixes inconsistent spacing for params in mpl drawer

### LINKED ISSUE #4845: Param display with pi's in mpl drawer inconsistent due to latex_to_text
### Information

- **Qiskit Terra version**: master
- **Python version**: 3.8
- **Operating system**: Ubuntu 18.04

### What is the current behavior?

The return from the latex_to_text call when the text comes from params that went through pi_check and contain pi is odd. Here are the ascii/unicode values from the return from the call with `text = $-$$\pi$/2, $-$$\pi$/2, $-$$\pi$/2`

`45 10 32 32 32 32 960 47 50 44 45 10 960 47 50 44 32 45 10 32 32 32 32 960 47 50`

Note the 10s (linefeed) and 32s (spaces). These seem to be arbitrarily thrown into the string and are somewhat random. No idea why. This affects the text width calc since this adds width. Note the 3rd gate below.

In addition the math mode minus sign is about twice the width of the hyphen that is used in pi_check, and is about the same width as the + sign. So replacing the hyphen with a + for the width calc will help as well.

Finally, the character that replaces non-ascii chars such as pi is a bit too small. Hence the tight spacing on the 2nd and 4th gates below.
image

### Steps to reproduce the problem

```
from qiskit import QuantumCircuit
from numpy import pi

qc = QuantumCircuit(1)
qc.u3(-3*pi/2, -3*pi/2, -3*pi/2, 0)
qc.u3(3*pi/2, 3*pi/2, 3*pi/2, 0)
qc.u3(-pi/2, -pi/2, -pi/2, 0)
qc.u3(pi/2, pi/2, pi/2, 0)
qc.draw('mpl', filename='param_latex.png')
```

### Suggested solutions

The obvious thing to do is to remove the 10s and 32s, replace the hyphen with a +, and make the default char larger. Note that the 32s always follow a 10, so if we remove `10 32 32 32 32` first and then any additional 10s, it should not affect other text. Of course, since we don't know why this stuff is there, this could change in a future release of latex_to_text, but this shouldn't do any serious harm in that case. Doing this produces,
image

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fixes #4845

### Details and comments

This PR fixes several spacing issues for params in the mpl drawer which were highlighted in #4845. These affect the text width calcs for params.
- Removes linefeeds and spaces from the output of latex_to_text
- Replaces hyphens with math mode minus signs in computing text width
- Adjusts to a slightly larger default char to cover non-ascii chars such as pi and other Greek letters

Converts this output
image
to
image

### CODE REVIEW

**Comment by 1ucian0** [comment]
mybinder test https://mybinder.org/v2/gh/enavarro51/qiskit-terra/fixissue4845branch?urlpath=apps/test/ipynb/mpl_tester.ipynb

File: qiskit/visualization/matplotlib.py [hunk 1]
```diff
@@ -282,14 +283,23 @@ def _get_text_width(self, text, fontsize):
             if num_carets:
                 text = text.replace('^', '', num_carets)
 
+            # This removes some arbitrary \n's and spaces that are returned
+            # by latex_to_text when parameters contain pi. Not sure why this
+            # happens and may change in a future release of latex_to_text. It
+            # also changes hyphen to + to match width of math mode minus sign.
+            if param:
+                text = text.replace('\n    ', '')
+                text = text.replace('\n', '')
```
Comment (1ucian0) [inline qiskit/visualization/matplotlib.py:292]:
```suggestion
                text = text.rstrip()
```

Comment (enavarro51) [reply qiskit/visualization/matplotlib.py]: Actually rstrip doesn't work. If you look at #4845, you'll see the 10's and 32's are mixed in the string.

**Review by 1ucian0 — APPROVED** [review]
Thanks!

**Pull request merged**